### PR TITLE
[#11235] datetime picker fix

### LINKED ISSUE #11235: Instructor audit logs: ensure end date is not before start date
Page: `/web/instructor/logs`

Currently, the date picker for 'search period until' allows you to input a date that is before the date from 'search period from'. It throws an error only after pressing the search button:

Screenshot 2021-06-29 at 12 23 43 AM

Not too urgent as the functionality is not affected. It seems to have a rather easy fix, by changing the min date option for the date picker

### PR DESCRIPTION
**Pull request by ChanwooJung1**
Fixes #11235

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.ts [hunk 1]
```diff
@@ -136,6 +136,32 @@ export class InstructorAuditLogsPageComponent implements OnInit {
...
+    if (this.isDateAfter(this.formModel.logsDateFrom, this.formModel.logsDateTo)) {
+      return false;
+    }
+    else if (this.isDateSame(this.formModel.logsDateFrom, this.formModel.logsDateTo)) {
+      return this.isTimeAfter(this.formModel.logsTimeTo, this.formModel.logsTimeFrom);
+    }
+    else {
+      return true;
+    }
+  }
```
Comment (daongochieu2810) [inline src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.ts:164]: I dont think these util functions should be placed here. Lets move them to `src/web/app/components/datepicker/datepicker.component.ts` or create a new module called `date-utils` to handle datetime operations

**Review by daongochieu2810 — CHANGES_REQUESTED** [review]
Changes look good!

**Review by halfwhole — COMMENTED** [review]
It looks like the current approach relies on disabling search if the date is out of range, but perhaps we could restrict the date picker more directly?

https://github.com/TEAMMATES/teammates/blob/69ae1cc91e6d5d8b39a91be88022f80618f8f39a/src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.html#L57

There's a `minDate` option in this line that could be changed to say `formModel.logsDateFrom`, would this solve our needs?

**Comment by hhdqirui** [comment]
@ChanwooJung1 for your kind update on the above.

**Comment by halfwhole** [comment]
@ChanwooJung1 a kind reminder on this PR! please do tell us how you wish to proceed, thanks

**Comment by damithc** [comment]
Guys, This PR seems to be stalling (no activities for the past 7 days). :snail: :cry:
Hope someone can get it to move forward again soon...

**Comment by daongochieu2810** [comment]
@ChanwooJung1 any update? If not we might have to close this PR and let other people take on the issue

**Comment by moziliar** [comment]
Close due to inactivity and open the issue up for other first-timers. Feel free to reopen and start contributing if there isn't already an active contributor for the issue!

**Pull request closed without merging**